src/components/course/CourseRunSelector.jsx

```diff
@@ -0,0 +1,50 @@
+import React, { useContext, useState, useRef } from 'react';
+import moment from 'moment';
+
+import { CourseContext } from './CourseContextProvider';
+import { SET_COURSE_RUN } from './data/constants';
+
+import './styles/CourseRunSelector.scss';
+
+export default function CourseRunSelector() {
+    const { dispatch, state } = useContext(CourseContext);
+    const {
+        activeCourseRun,
+        course: {
+            courseRuns,
+        },
+    } = state;
+    const [multipleRunsAvailable, setMultipleRunsAvailable] = useState(courseRuns.length > 1);
+    const [editing, setEditing] = useState(false);
+    const selectRef = useRef(activeCourseRun.uuid);
+
+    const handleClick = () => {
+        const selectedCourseRun = courseRuns.find((courseRun) => courseRun.uuid === selectRef.current.value);
```

Comment:
Seemed better to do this find here rather than store the courseRun as the value on a select

Reply:
Curious - why would it be better to use a find than to store the courseRun on select as a state?

Reply:
Results in the following HTML which seemed cleaner than a JSON.stringify form of the object.
```
<select id="course-run-select">
  <option value="ce9618e0-5131-4486-8d9f-84528238916a">
    Feb 5, 2013</option>
  <option value="faaabc10-60fd-467c-a41f-6fb22af59b87">
    Dec 31, 2029
  </option>
</select>
```
Would have been happy to us the plain JS object which results in this, but won't allow you to reference it later as the ref returns a string.
```
<select id="course-run-select">
  <option value="[object Object]">
    Feb 5, 2013
  </option>
  <option value="[object Object]">
    Dec 31, 2029
  </option>
</select>
```

Alternatively I could have done the index in the array as the value rather than the uuid, but felt uuid was more intuitive.

Reply:
hmmm I might just be thinking of some wrappers around dropdowns that I used with react-bootstrap. It made it a little easier to just pass the values around. But at the same time it's a little harder to customize/is abstracted away.